packages/pluggableWidgets/screenshotrunner-native/src/ScreenshotRunner.tsx

```diff
@@ -0,0 +1,158 @@
+import { createElement, ReactElement, useState, Fragment, useEffect, useCallback } from "react";
+import { executeAction } from "@mendix/piw-utils-internal";
+import { FlatList, View, Text, ListRenderItemInfo, Pressable } from "react-native";
+import { flattenStyles } from "@mendix/piw-native-utils-internal";
+import { ScreenshotRunnerProps, TasksType } from "../typings/ScreenshotRunnerProps";
+import { defaultScreenshotRunnerStyles, ScreenshotRunnerStyle } from "./ui/Styles";
+
+declare type ErrorType = {
+    name: string;
+    error: Error;
+};
+declare type CustomGlobal = NodeJS.Global &
+    typeof globalThis & {
+        screenshotRunner: {
+            masterIsRunning: boolean;
+            task?: {
+                name: string;
+                isRunning: boolean;
+                error: boolean;
+                onError: (e: Error, name: string) => any;
+            };
+        };
+    };
+
+const timeout = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+export function ScreenshotRunner(props: ScreenshotRunnerProps<ScreenshotRunnerStyle>): ReactElement {
+    const [tasksRunning, setTasksRunning] = useState<TasksType[]>([]);
+    const [taskWithErrors, setTaskWithErrors] = useState<ErrorType[]>([]);
+    const [areTasksTriggered, setAreTasksTriggered] = useState(false);
+    const styles = flattenStyles(defaultScreenshotRunnerStyles, props.style);
+    const { androidRipple: androidRippleStart } = styles.button.start.container;
+    const { androidRipple: androidRippleStop } = styles.button.stop.container;
+    const tasksEnabledLength = props.tasks.filter(task => task.enabled).length;
+
+    const onError = (e: Error, name: string) => {
+        console.error(e);
+        setTaskWithErrors(oldArray => [
+            ...oldArray,
+            {
+                name,
+                error: e
+            }
+        ]);
+    };
+    const enableGlobalScreenshotVariable = (name = "placeholder") => {
+        (global as CustomGlobal).screenshotRunner = {
+            masterIsRunning: true,
+            task: {
+                name,
+                isRunning: true,
+                error: false,
+                onError: (e: Error) => {
+                    onError(e, name);
+                }
+            }
+        };
+    };
+
+    const runTask = useCallback(async (task: TasksType) => {
+        const maxTimeouts = 30;
+        let timeoutCount = 0;
+
+        if (!task.task?.canExecute) {
+            throw new Error(`Task ${task.name} is not executable..`);
+        }
+
+        setTasksRunning(oldArray => [...oldArray, task]);
+        enableGlobalScreenshotVariable(task.name);
+        executeAction(task.task);
+
+        while ((global as CustomGlobal).screenshotRunner?.task?.isRunning && timeoutCount <= maxTimeouts) {
```

Comment:
I would write a `getScreenshotRunner` (and `setScreenshotRunner` above) instead of have these `(global as CustomGlobal)` casts everywhere in the code. Then you also need typings only for the screenshot runner itself, not the entire custom global object, and abstract away where it's stored.

It's also not clear to me why this needs to be in the global object and cannot be done with `useState()`, but maybe I'll understand more when I dig deeper into the code :).